### ISSUE TITLE
`vue/component-name-in-template-casing` compatible auto import components

### ISSUE THREAD

**Issue by momei-LJM** [vuejs/eslint-plugin-vue#2762]
**What rule do you want to change?**
If I want to directly use the components imported by `unplugin-vue-components `, I need to manually list all the components in the `globals` section.

Is it possible to make the paths of the automatically imported `.dts` files compatible with the `globals` and to extract the automatically imported components?

**Does this change cause the rule to produce more or fewer warnings?**
no

**How will the change be implemented? (New option, new default behavior, etc.)?**
`globals` supports the `.dts` path and then parses out the internal components.

**Please provide some example code that this change will affect:**

```vue

```

**What does the rule currently do for this code?**

**What will the rule do after it's changed?**

**Additional context**

**Comment by ota-meshi** [comment]
I think you should try `registeredComponentsOnly: false`.

**Comment by momei-LJM** [comment]
> I think you should try `registeredComponentsOnly: false`.

It works very well. I just think this option might be a bit too arbitrary, huh... Ha ha😂
